### ISSUE TITLE
[FEATURE] "HEAD_ITEM"

### ISSUE THREAD

**Issue by Anfauglir-Jakob** [Blancworks/Figura#139]
It would be very cool to have something like RIGHT_HELD_ITEM but for the head since players can wear items in the head slot that aren't armor and it would be nice to be able to change their size and position and stuff.

**Comment by Francy-chan** [comment]
its different than `armor_model.HEAD_ITEM` because `RIGHT_HELD_ITEM` is a blockbench keyword, which scales and is anchored on their parent part

currently theres no way to scale the armor_model

**Comment by Anfauglir-Jakob** [comment]
Yes, minecraft armor is a problem. That's why this request suggests a new blockbench keyword to be added for items like a pumpkin or anything else that could be put in the head slot, since these should not have the same limitations as the armor I think.

**Comment by Anfauglir-Jakob** [comment]
Since vanilla_model, as well as armor resizing, is now possible I want to bring this up again. ^^"

Minecraft allows using any item in your head_slot, it doesn't have to be a helmet. (ex. Pumpkin, but every item works)
It even has its own display settings for the head if you create an item(_model).
This is used by many to give players custom helmets and other deco elements for their heads and would be really great if Figura would be able to display them correctly.

As of right now, it seems like they don't resize in any form and couldn't be moved around.
I can imagine it could be possible to make them work like both of the hand items work in Figura but since I'm not a good programmer I cannot be sure or push it myself. :/

Right now it's the only thing that prevents me from publishing a revised version of the resize script because I cannot get the pumpkin (and other head items like helmets and horns) to work correctly with the script.

**Comment by GrandpaScout** [comment]
You could probably just add the item back by using the
`renderer.renderItem(ItemStack, ModelPart, TransformationMode, emissive bool, pos, rot, scale)`
function.

It can be used like so,
```lua
renderer.renderItem(
  player.getEquipmentItem(6),
  model.Head,
  "HEAD",
  false,
  {x_pos, y_pos, z_pos},
  {x_rot, y_rot, z_rot},
  {x_scale, y_scale, z_scale}
)
```
to recreate the head item effect.

Because this uses the head item directly, all tags go along with it (including `CustomModelData`.)

**Comment by Anfauglir-Jakob** [comment]
I'll give this a try as soon as I can
Thank you very much! :3

Nonetheless, it would be great if you don't have to do this manually since it's the same mechanic the hands use too.
But if your method works, this is just a quality of life feature... ^^
